lib/screens/scan_screen.dart

```diff
@@ -0,0 +1,166 @@
+import 'dart:convert';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:scan/scan.dart';
+import 'package:http/http.dart' as http;
+import 'package:image_picker/image_picker.dart';
+
+// ignore: must_be_immutable
+class ScanScreen extends StatefulWidget {
+  dynamic data;
+  ScanScreen({Key? key, required this.data}) : super(key: key);
+  @override
+  _ScanScreenState createState() => _ScanScreenState();
+}
+
+class _ScanScreenState extends State<ScanScreen> {
+  ScanController controller = ScanController();
+  late int _nowTime;
+  final DateTime _now = DateTime.now();
+
+  Future<void> _showAlert(String title, String message) async {
+    return showDialog<void>(
+      context: context,
+      barrierDismissible: false, // user must tap button!
+      builder: (BuildContext context) {
+        return CupertinoAlertDialog(
+          title: Text(title),
+          content: Text(message),
+          actions: <Widget>[
+            CupertinoDialogAction(
+              child: const Text('Okay'),
+              onPressed: () {
+                Navigator.of(context).pop();
+              },
+            ),
+          ],
+        );
+      },
+    );
+  }
+
+  // Called upon scan of QrCode from the 3 options
+  validate(String deviceId, String seed) async {
+    final url = Uri.parse('http://localhost:8080/user_val').replace(queryParameters: {'id': deviceId, 'seed': seed});
+    var response = await http.get(url);
```

Comment:
suggestion: Ideally, this logic would be handled by your state management in a separate class